### ISSUE TITLE
Mandatory fields in ErrorHandler - Maximum redeliveries and Redelivery delay

### ISSUE THREAD

**Issue by tplevko** [KaotoIO/kaoto-next#356]
### Describe the Bug

This is more a question regarding the  Maximum redeliveries and Redelivery delay - shouldn't these fields be marked as mandatory in the ErrorHandler?

### Steps to Reproduce the Bug or Issue

1. see the log errorHandler configuration

### Screenshots or Videos

![Screenshot from 2023-11-16 16-23-44](https://github.com/KaotoIO/kaoto-next/assets/4180208/2bf7e976-53bc-4f1a-92c0-593e320c3b9f)

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [latest]

**Comment by lordrip** [comment]
sink error handler

![Image](https://github.com/KaotoIO/kaoto-next/assets/16512618/a8ab59b7-e933-44cf-9b6e-ad68a8dfb616)

**Comment by tplevko** [comment]
I conducted a research and actually haven't found any information in camel documentation regarding the  `Maximum redeliveries` and `Redelivery delay` being mandatory. I think, this issue can be closed with `wontfix`

**Comment by tplevko** [comment]
Sure, no problem, was just sharing my findings, but if there is a reason to have this in place, we can of course leave this open.

**Comment by igarashitm** [comment]
### tl;dr
These parameters are optional.

### Details
The Pipe log ErrorHandler is eventually built as a `DefaultErrorHandler`. When I deploy following with `kubectl -f a.pipe.yaml`
```yaml
apiVersion: camel.apache.org/v1
kind: Pipe
metadata:
  name: pipe-2327
spec:
  source:
    properties:
      message: hello
    ref:
      apiVersion: camel.apache.org/v1
      kind: Kamelet
      name: timer-source
  sink:
    ref:
      apiVersion: camel.apache.org/v1
      kind: Kamelet
      name: log-sink
  errorHandler:
    log:
      parameters: {}
```

Then I can see it's assigned to `DefaultErrorHandler` by `kamel describe integration pipe-2327` 
```
...
Configuration:
  Type:	property
  Value:	camel.beans.defaultErrorHandler = #class:org.apache.camel.builder.DefaultErrorHandlerBuilder
  Type:	property
  Value:	camel.k.errorHandler.ref = defaultErrorHandler
...
```

So `redeliveryDelay` and `maximumRedeliveries are assigned through 
- https://github.com/apache/camel/blob/main/core/camel-core-model/src/main/java/org/apache/camel/model/errorhandler/DefaultErrorHandlerDefinition.java#L438
- https://github.com/apache/camel/blob/main/core/camel-core-model/src/main/java/org/apache/camel/model/errorhandler/DefaultErrorHandlerDefinition.java#L448

On the other hand, since `DefaultErrorHandler` itself and redelivery policy don't have `required` field, in theory all the options are optional.
- https://github.com/apache/camel/blob/main/dsl/camel-yaml-dsl/camel-yaml-dsl/src/generated/resources/schema/camelYamlDsl.json#L12144
- https://github.com/apache/camel/blob/main/dsl/camel-yaml-dsl/camel-yaml-dsl/src/generated/resources/schema/camelYamlDsl.json#L4202

BUT,

### More parameters exist on log and sink errorHandler
Although at first I added only these 2 parameters into the schema by looking at this example
https://camel.apache.org/camel-k/next/kamelets/kameletbindings-error-handler.html

In fact the `log` errorHandler is built with `DefaultErrorHandlerBuilder`, and also the `sink` errorHandler is built with `DeadLetterChannelBuilder` , therefore all the parameters these builder support could be specified. We need to update the schema and add those parameters. Note that there's no official schema for Pipe ErrorHandler. Because of that we're using a custom made one in Kaoto. Unfortunately it's even different from `DefaultErrorHandlerDefinition` in camelYamlDsl, for example `redeliveryDelay` is not a direct parameter but is specified through redelivery policy in camelYamlDsl.

**Comment by lhein** [comment]
outdated, closing
